images/checkpatch/checkpatch.sh

```diff
@@ -36,6 +36,7 @@ ignore_list=(
     # Checks
     BIT_MACRO
     LONG_LINE_COMMENT
+    MACRO_ARG_REUSE
```

Comment:
CI test still fails even if it is not displayed in red. Could you tell me how to deal with this in a case-by-case basis. All I could come up with was to ignore this check for the whole cilium/cilium repo.

Reply:
This was ignored due to the macro really needing to to use the argument multiple times, so in this case it is a false positive.

Reply:
IMO #2 above would be the best solution, basically allow a comment right above the offending line basically turning on `--ignore` options on a case-by-case basis. This would be some feature work on checkpatch, no?

#3 could work if we had a way for the CI to observe PR comments, pick out checkpatch args from there, and pass them to the CI run locally. `bpf/Makefile.bpf` already passes `CHECKPATCH_ARGS` environment variable to checkpatch, so the only thing missing is picking them up from PR comments. This would have the downside that the same checkpatch failure can happen on the same code also later, in unrelated PRs,  which would be really annoying, especially when the PR author has no idea how to suppress it.

On #4 we have recently had discussion about the use of the `ready-to-merge` label in the project, with the conclusion that it may only be set once CI is green for all required tests.

So in general I agree, as long as we do not have one of 1-3 above impelemented, we should probably mark checkpatch CI run as not required to not block work on the project.